test/trento/application/usecases/heartbeats_test.exs

```diff
@@ -24,32 +23,35 @@ defmodule Trento.HeartbeatsTest do
     now = DateTime.utc_now()
     agent_id = Faker.UUID.v4()
 
-    with_mock DateTime, [:passthrough], utc_now: fn -> now end do
-      expect(
-        Trento.Commanded.Mock,
-        :dispatch,
-        fn command ->
-          assert %UpdateHeartbeat{
-                   host_id: ^agent_id,
-                   heartbeat: :passing
-                 } = command
-
-          :ok
-        end
-      )
+    expect(
+      Trento.Support.DateService.Mock,
+      :utc_now,
+      fn -> now end
```

Comment:
I found [this article](https://codeincomplete.com/articles/testing-dates-in-elixir/) that deals with dates and tests.
What do you think?

Reply:
I find it more complex and difficult to follow. You still need to have a mocked functionality, that is hidden within that service with a bunch of helper methods for testing. It looks like it tries to create its own mocking system. And you cannot use tools like Mox

The best thing about the current solution is that it is straightforward and doesn't require and configuration. And the tests are super simple to understand.

Some refs: https://blog.carbonfive.com/lightweight-dependency-injection-in-elixir-without-the-tears/

Reply:
Fair, I am fine with this approach.